extensions/src/main/scala/ai/h2o/sparkling/extensions/rest/api/InternalUtilsServlet.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.extensions.rest.api
+
+import ai.h2o.sparkling.extensions.rest.api.schema.ScalaCodeResult
+import ai.h2o.sparkling.utils.ScalaUtils.withResource
+import com.google.gson.Gson
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+import water.H2O.H2OCountedCompleter
+import water.api.schemas3.JobV3
+import water.server.ServletUtils
+import water.{DKV, Job, Key}
+
+import scala.collection.concurrent.TrieMap
+
+/**
+  * This servlet class handles POST requests for the path /3/sw_internal
+  */
+class InternalUtilsServlet extends ServletBase {
+
+  private val jobs = new TrieMap[String, Boolean]
+
+  case class IdParameter(id: String)
+
+  object IdParameter {
+    def parse(request: HttpServletRequest): IdParameter = {
+      val jobId = request.getRequestURI.split("/")(4)
+      IdParameter(jobId)
+    }
+  }
+
+  override def doPost(request: HttpServletRequest, response: HttpServletResponse): Unit = {
+
+    processRequest(request, response) {
+      val obj = request.getRequestURI match {
+        case s if s.startsWith("/3/sw_internal/start/") =>
+          val parameters = IdParameter.parse(request)
+          val key = parameters.id
+          val job =
+            new Job[ScalaCodeResult](
+              Key.make[ScalaCodeResult](key),
+              classOf[ScalaCodeResult].getName,
+              "ScalaCodeResult")
+          val jobV3 = new JobV3(job)
+          jobs.put(jobV3.key.name, true)
+          job.start(new H2OCountedCompleter() {
+            override def compute2(): Unit = {
+              while (jobs.contains(jobV3.key.name) && jobs(jobV3.key.name)) {
+                Thread.sleep(100)
```

Comment:
Why do we need `Thread.sleep(100)`?

Reply:
I have tried to document this in the code:

``
/**
 * In Asynchronous mode of Scala cell execution H2O Flow expects that the cell returns H2O Job right away and then
 * when clicking on the job link, the user will either get informed that the job is still running or the result will
 * be displayed.
 *
 * We need to perform the computation on Spark Driver to have access to both H2O and Spark and since H2O expects the H2O
 * Job object, we are submitting dummy H2O job which is alive for the duration of the computation on the Spark side.
 *
 * That is the reason for submitDummyWrapperH2OJob and stopDummyWrapperH2OJob methods.
 */
``

That is also the reason for the sleep. We need to wait until thread on Spark driver finishes before we can mark this dummy job as stopped.